File: src/main/java/com/etour/app/entity/DepartureDateMaster.java
package com.etour.app.entity;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "departure_date_master")
public class DepartureDateMaster {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "departure_date_id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "catmaster_id", nullable = false)
    private CategoryMaster catmaster;

    @Column(name = "departure_date", nullable = false)
    private LocalDate departureDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    @Column(name = "number_of_days", nullable = false)
    private Integer numberOfDays;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public CategoryMaster getCatmaster() {
        return catmaster;
    }

    public void setCatmaster(CategoryMaster catmaster) {
        this.catmaster = catmaster;
    }

    public LocalDate getDepartureDate() {
        return departureDate;
    }

    public void setDepartureDate(LocalDate departureDate) {
        this.departureDate = departureDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDate endDate) {
        this.endDate = endDate;
    }

    public Integer getNumberOfDays() {
        return numberOfDays;
    }

    public void setNumberOfDays(Integer numberOfDays) {
        this.numberOfDays = numberOfDays;
    }

}

File: dotnet/backend/Models/departure_date_master.cs
using System;
using System.Collections.Generic;

namespace Etour_Backend_dotnet.Models;

public partial class departure_date_master
{
    public int departure_date_id { get; set; }

    public int catmaster_id { get; set; }

    public DateOnly departure_date { get; set; }

    public DateOnly end_date { get; set; }

    public int number_of_days { get; set; }

    public virtual ICollection<booking_header> booking_header { get; set; } = new List<booking_header>();

    public virtual category_master catmaster { get; set; } = null!;

    public virtual ICollection<tour_master> tour_master { get; set; } = new List<tour_master>();
}
